Realm/RLMSyncSessionRefreshHandle.mm

```diff
@@ -93,29 +107,60 @@ - (void)scheduleRefreshTimer:(NSDate *)dateWhenTokenExpires {
 /// Handler for network requests whose responses successfully parse into an auth response model.
 - (BOOL)_handleSuccessfulRequest:(RLMAuthResponseModel *)model strongUser:(RLMSyncUser *)user {
     // Success
-    if (auto session = _session.lock()) {
-        if (session->state() != SyncSession::PublicState::Error) {
-            session->refresh_access_token([model.accessToken.token UTF8String], none);
-            NSDate *expiration = [NSDate dateWithTimeIntervalSince1970:model.accessToken.tokenData.expires];
-            [self scheduleRefreshTimer:expiration];
-            return YES;
+    std::shared_ptr<SyncSession> session = _strongSession ?: _session.lock();
```

Comment:
I think this can just be `session = _session.lock()`, since `_strongSession` only exists to ensure the session remains alive until we enter this method (or the corresponding failure method).